retry.go

```diff
@@ -0,0 +1,67 @@
+package squalor
+
+import (
+	"fmt"
+	"github.com/go-mysql/conn"
+	"github.com/go-sql-driver/mysql"
+	"net"
+	"time"
+)
+
+type RetryConfiguration struct {
+	Retries                     int
+	SleepBetweenRetriesInMillis int64
+	RetryableExceptions         map[error]bool
+}
+
+var BasicRetryConfiguration = RetryConfiguration{
+	3,
+	300,
+	map[error]bool{conn.ErrReadOnly: true, conn.ErrConnLost: true, conn.ErrConnCannotConnect: true, conn.ErrTimeout: true, mysql.ErrInvalidConn: true}}
+
+var NoOpRetryConfiguration = RetryConfiguration{
+	0,
+	0,
+	map[error]bool{}}
+
+func Retry(retryConfig RetryConfiguration, fn func() error) error {
```

Comment:
Hmm interesting, It's clean to specify the Retryable interface function that way. I changed it so (but I am a bit unfamiliar with Golang so am grateful for your suggestions.) For your last statement, I do want to give clients an easy way to tweak the 3 parameters we have so that they can change as they see fit rather than having to duplicate the whole Retryable function just to change the number of retries, etc.